.github/workflows/ci.yml

```diff
@@ -0,0 +1,375 @@
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          make lint.install
+      - name: Check
+        run: |
+          make lint.check
+  build:
+    name: Build API
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - id: "auth"
+        if: github.repository_owner == 'codecov'
+        name: "Authenticate to Google Cloud"
+        uses: "google-github-actions/auth@v1.1.1"
+        with:
+          token_format: "access_token"
+          workload_identity_provider: ${{ secrets.CODECOV_GCP_WIDP }}
+          service_account: ${{ secrets.CODECOV_GCP_WIDSA }}
+
+      - name: Docker configuration
+        if: github.repository_owner == 'codecov'
+        run: |-
+          echo ${{steps.auth.outputs.access_token}} | docker login -u oauth2accesstoken --password-stdin https://us-docker.pkg.dev
+
+      - name: Cache Requirements
+        id: cache-requirements
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-api-requirements
+        with:
+          path: |
+            requirements.tar
+          key: ${{ runner.os }}-api-${{ env.cache-name }}-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/docker/Dockerfile.requirements') }}
+          restore-keys: |
+            ${{ runner.os }}-api-${{ env.cache-name }}-
+
+      - name: Cache App
+        id: cache-app
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-api-app
+        with:
+          path: |
+            app.tar
+          key: ${{ runner.os }}-api-${{ env.cache-name }}-${{ github.run_id }}
+
+      - name: Load requirements from cache
+        if: ${{ steps.cache-requirements.outputs.cache-hit == 'true' }}
+        run: |
+          docker load --input requirements.tar
+
+      - name: Build/pull requirements
+        if: ${{ steps.cache-requirements.outputs.cache-hit != 'true' }}
+        run: |
+          make build.requirements
+          make save.requirements
+
+      - name: Push Requirements
+        if: ${{ steps.cache-requirements.outputs.cache-hit != 'true' && github.repository_owner == 'codecov' }}
+        run: |
+          make push.requirements
+      - name: Build app
+        run: |
+          make build.app
+          make save.app
+
+  codecovstartup:
+    name: Codecov Startup
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Install CLI
+        run: |
+          make test_env.install_cli
+      - name: Run Startup
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_ORG_TOKEN }}
+        run: |
+          codecovcli create-commit
+          codecovcli create-report
+      - name: Run Startup Staging
+        run: |
+          codecovcli create-commit -t ${{ secrets.CODECOV_ORG_TOKEN_STAGING }}
```

Comment:
I see this was just copied from the CircleCI config but I'm surprised this doesn't need a `--url` argument to point to the staging API.

Reply:
It does, tho... or it should at least...

Reply:
Probably best to turn `--fail-on-error` on in everything
https://github.com/codecov/codecov-api/actions/runs/6140828497/job/16660340146

```
info - 2023-09-11 01:37:19,169 -- ci service found: github-actions
info - 2023-09-11 01:37:19,312 -- Process Commit creating complete
error - 2023-09-11 01:37:19,313 -- Commit creating failed: {"detail":"Invalid token."}
info - 2023-09-11 01:37:19,670 -- ci service found: github-actions
info - 2023-09-11 01:37:19,791 -- Process Report creating complete
error - 2023-09-11 01:37:19,791 -- Report creating failed: {"detail":"Invalid token."}
```

Reply:
Added url and fail on error.